workspaces/homepage/plugins/dynamic-home-page/src/api/ModelCatalogApiClient.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createApiRef } from '@backstage/core-plugin-api';
+import {
+  CatalogApi,
+  EntityFilterQuery,
+  GetEntitiesRequest,
+  GetEntityFacetsRequest,
+} from '@backstage/catalog-client';
+import type { Entity } from '@backstage/catalog-model';
+
+export interface QueryEntitiesResponse<T> {
+  items: T[];
+  totalItems: number;
+  pageInfo: {
+    nextCursor?: string;
+    prevCursor?: string;
+  };
+}
+
+export interface ModelCatalogApi {
+  getModels(
+    request: GetEntitiesRequest,
+  ): Promise<QueryEntitiesResponse<Entity>>;
+  getTemplates(
+    request: GetEntitiesRequest,
+  ): Promise<QueryEntitiesResponse<Entity>>;
+}
+
+export const modelCatalogApiRef = createApiRef<ModelCatalogApi>({
+  id: 'app.developer-hub.model-catalog.service',
+});
+
+export class ModelCatalogApiClient implements ModelCatalogApi {
+  private readonly catalogApi: CatalogApi;
+
+  constructor(options: { catalogApi: CatalogApi }) {
+    this.catalogApi = options.catalogApi;
+  }
+
+  appendFilter = <T extends GetEntitiesRequest | GetEntityFacetsRequest>(
+    request: T,
+    filter: EntityFilterQuery,
+  ): T => ({
+    ...request,
+    filter: {
+      ...request.filter,
+      ...filter,
+    },
+  });
+
+  async getModels(
+    request: GetEntitiesRequest,
+  ): Promise<QueryEntitiesResponse<Entity>> {
+    const response = await this.catalogApi.queryEntities(
+      this.appendFilter(request, {
+        kind: 'Resource',
```

Comment:
This will only return `Resource` entities but "Explore your software catalog" section talks about browsing Components, Resources, Systems and APIs in your organisation.

Can you please make this an array with other kinds as well ?

```suggestion
        kind: [ 'Component', 'API', 'Resource', 'System'],
```